### PR TITLE
Tweaks poi-preset-karma

### PR DESCRIPTION
**Pull request by egoist**
/cc @mblarsen

### CODE REVIEW

File: packages/poi-preset-karma/README.md [hunk 1]
```diff
@@ -100,9 +100,21 @@ This preset can also directly read Karma config from `karma` property in `poi.co
...
-Then the `frameworks` we finally got would be `['mocha', 'chai']`.
+We use `Object.assign` to merge custom karma config with our default one.
+
+`karma` could also be a function, then we will use its return value as karma config:
+
+```js
+// poi.config.js
+module.exports = {
+  karma(config) {
+    config.reporters = ['nyancat']
```
Comment (mblarsen) [inline packages/poi-preset-karma/README.md]: Hehehe

**Comment by mblarsen** [comment]
This doesn't solve the bit odd thing that reporters will have mocha added no matter what unless you override it using the karma settings, where as it will always be overwritten if you use frameworks.

**Comment by mblarsen** [comment]
Maybe make the mocha reporter also just a default not a forced one, just so they behave the same.

**Comment by egoist** [comment]
@mblarsen are you interested in becoming a collaborator? 😄 so you can update my pr   more easily

**Comment by mblarsen** [comment]
I could do that. So you want me to make the suggested changes to this PR, right?

**Comment by egoist** [comment]
@mblarsen yeah I don't have my computer now and more importantly I don't really use karma 😂

**Comment by mblarsen** [comment]
@egoist from commit message:

`reporters` default to `[mocha]`. If other reporters are set in the
preset options they replace the default. In either case `coverage`
will be added based on the coverage setting (`coverage: true` or
`--coverage` on the command line).

If you want to role your own coverage reporting the `karma:` poi
options completely override the preset settings.

--

I've tested it in a bunch of combinations setting reporters in preset options and en poi karma options. 

Didn't update the docs yet.

File: packages/poi-preset-karma/package.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "name": "poi-preset-karma",
   "version": "9.1.0",
   "scripts": {
-    "example": "poi test"
+    "example": "poi test",
+    "example:coverage": "poi test --coverage"
```
Comment (egoist) [inline packages/poi-preset-karma/package.json]: `yarn example --coverage` 😄

Comment (mblarsen) [reply packages/poi-preset-karma/package.json]: Removed again :)

**Comment by mblarsen** [comment]
Looks okay to merge? Will build the ts stuff on top of this. Easier to merge than if I branch from master :) @egoist

**Pull request merged**